armory/eval/evaluator.py

```diff
@@ -24,6 +25,42 @@
 logger = logging.getLogger(__name__)
 
 
+def tmp_output_subdir(retries=10):
+    """
+    Return (<subdir name>, <tmp subdir path>, <output subdir path>)
+
+    retries - number of times to retry folder creation before returning an error
+        if retries < 0, it will retry indefinitely.
+    """
+    tries = int(retries) + 1
+    host_paths = paths.host()
+    while tries:
+        subdir = datetime.datetime.utcnow().isoformat()
+        # ":" characters violate docker-py volume specifications
+        subdir = subdir.replace(":", "-")
+        # Use tmp_subdir for locking
+        try:
+            tmp_subdir = os.path.join(host_paths.tmp_dir, subdir)
+            os.mkdir(tmp_subdir)
+        except FileExistsError:
+            tries -= 1
+            if tries:
+                logger.warning(f"Failed to create {tmp_subdir}. Retrying...")
+            continue
+
+        try:
+            output_subdir = os.path.join(host_paths.output_dir, subdir)
+            os.mkdir(output_subdir)
+        except OSError:
+            logger.exception(
+                f"Sync failure: created {tmp_subdir} but failed on {output_subdir}. Exiting."
+            )
+            raise
```

Comment:
Why are we catching an OSError here... whats the situation that occurs in? Do we need to do clean-up of the `tmp_subdir`?

Reply:
This may not be something that we need to worry about, as I can't think of why this would occur. I think it would be fine to not catch and then have it die.

Reply:
Yeah I think it's safe to remove this catch.

Reply:
Removed.